src/containers/Register/RegisterForm.tsx

```diff
@@ -1,14 +1,84 @@
-import React from "react";
+import React, {useEffect} from "react";
 import {Input} from "../../components/Input/Input";
 import {Button} from "../../components/Button/Button";
+import {useForm, Controller} from "react-hook-form";
 
-export const RegisterForm = () => {
-    return (
-        <form>
-            <Input placeholder="Email" icon="email" />
-            <Input placeholder="Password" type="password" icon="lock" />
-            <Input placeholder="Repeat password" type="password" icon="lock" />
+export interface IRegisterFormProps {
+    onSubmit: (submitData: IRegisterFormData) => void;
+}
+
+export interface IRegisterFormData {
+    email: string;
+    password: string;
+    repeatPassword: string;
+}
+
+export const RegisterForm = (props: IRegisterFormProps) => {
+    const {register, handleSubmit, errors, control, watch} = useForm<IRegisterFormData>();
 
+    const onSubmit = handleSubmit((submitData: IRegisterFormData) => {
+        if (submitData.password === submitData.repeatPassword) props.onSubmit(submitData);
+        else console.log("Passwords don't match!");
```

Comment:
why console.log? I don't think you need this check here as handleSubmit is called only after validation has passed